pc2cps

```diff
@@ -72,27 +92,76 @@ class pkgconfig(object):
-            return None if x is None else x.keys()
-
-        c = {
-            'Type': 'interface',
-        }
-        extend(c, 'Link-Flags', self._get_split('Libs'))
-        extend(c, 'Requires', _keys(self.requires))
+            return [] if x is None else x.keys()
+
+        def _list(re_):
+            ma = [re_.match(x) for x in link_flags]
+            return [x.group(1) for x in ma if x]
+
+        # If not found and if package name given in library list with
+        # "-l" flag, an error should be thrown?
+        def _find_location():
+            for order in self.parse_order:
+                for path in paths:
+                    for file in os.listdir(path):
+                        pattern = '(lib)?' + self.library_name\
+                                  + self._lib_pattern[order]
+                        ma = re.match(pattern, file)
+                        if ma is not None:
+                            return os.path.join(path, ma.group(0))
+
+        # Preprocess libraries
+        link_flags = self._get_split('Libs')
+        # Find all library paths
+        paths = [x for x in _list(self._re_libpath) if os.path.isdir(x)]
+        # Find all libraries
+        self.libname = _list(self._re_libname)
+        _component = {}
+        # NOTE: We are making the assumption that the extension of the file is
+        # correct. This could be replaced by `os.path.realpath()` followed by
+        # calling the executable `file` to be safer.
+        location = _find_location()
+        if location:
+            # Extract library extension (=type)
+            all_patterns = '(' + self._lib_pattern['shared']\
+                           + '|'\
+                           + self._lib_pattern['static'] + ')'
+            ma = re.match('.*' + all_patterns, location)
+            if ma is not None:
+                # Expected format of ma.group(1): .so or .so.1 or .a or .dylib
+                # Remove potential .1 or other number past the actual extension
+                extension = ma.group(1).split('.')[1]
+                _component['Type'] = self._type[extension]
+                extend(_component, 'Location', location)
+                # Remove dependency on itself.
+                # some library names already contain the prefix `lib`.
+                self.libname = [x for x in self.libname
+                            if not re.match('(lib)?' + x, self.library_name)]
+        # Remove libraries and library paths from link flags as they are
+        # directly added to the requirements and the components.
+        link_flags = [x for x in link_flags if not x.startswith('-l')
+                      and not x.startswith('-L')]
+        extend(_component, 'Link-Flags', link_flags)
+        # Add system libraries found in link flags to requirements
+        list_requires = _keys(self.requires)
+        # Update requires dictionary
+        self.requires.update({x.lower(): {} for x in self.libname if x.lower()
```

Comment:
Upon further consideration, I do believe forcibly lower-casing these is broken. Package names, in both CPS and pkg-config, are not necessarily lower case, and CPS is (partially) case-sensitive.

Reply:
True. I had only seen packages with lower case but looking more I see that some pkg files do have uppercase. I'll update this.